src/Summoner/Tui/Kit.hs

```diff
@@ -0,0 +1,44 @@
+{-# LANGUAGE FlexibleInstances      #-}
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE MultiParamTypeClasses  #-}
+{-# LANGUAGE TemplateHaskell        #-}
+
+{- | This module contains data types to work with application form.
+'SummonKit' is the data type containing the values manipulated by the fields
+in the form.
+-}
+
+module Summoner.Tui.Kit
+       ( -- * Data types
+         SummonKit (..)
+       , initialSummonKit
+
+         -- * Lenses
+       , test1L
+       , test2L
+       ) where
+
+import Lens.Micro.TH (makeFields)
+
+import Summoner.Tui.CheckBox (CheckBox, toCheckBoxes)
+
+
+data Test1 = Test1 Text Text
+    deriving (Show)
+data Test2 = Test2 Text Int
+    deriving (Show)
+
+-- | Global TUI state.
+data SummonKit = SummonKit
+    { summonKitTest1L :: [CheckBox Test1]
+    , summonKitTest2L :: [CheckBox Test2]
```

Comment:
Later we can create our own lens rules to not write `L` suffix here manually :+1: